### PR TITLE
Add alert info in map

### PR DESCRIPTION
**Pull request by barshathakuri**
## Addresses:
- #22 

## Changes
- Add Alert info in Map

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/Home/AlertsView/AlertsAside/Admin1ListItem/i18n.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "namespace": "ongoingAlertRegion",
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/Admin1ListItem/i18n.json]: Let's use appropriate names

File: src/views/Home/AlertsView/AlertsAside/Admin1ListItem/i18n.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    "strings": {
+        "regionList": "Region List"
+    }
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/Admin1ListItem/i18n.json]: Let's use appropriate names.

File: src/views/Home/AlertsView/AlertsAside/Admin1ListItem/index.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import { Container } from '@ifrc-go/ui';
+import { useTranslation } from '@ifrc-go/ui/hooks';
+
+import { Admin1ListQuery } from '#generated/types/graphql';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type Admin1Type = NonNullable<NonNullable<Admin1ListQuery['public']>['admin1s']>['items'][number];
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/Admin1ListItem/index.tsx]: Lets rename `Admin1Type` to `Admin1`

File: src/views/Home/AlertsView/AlertsAside/Admin1ListItem/index.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+export interface RegionProps {
+    admin1WithAlert?: Admin1Type;
+}
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/Admin1ListItem/index.tsx]: Let's use appropriate names. Here `Props` would suffice.

File: src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type AletInfosType = NonNullable<NonNullable<AlertInfosQuery['public']>['alert']>;
+
+type InfoAlertType = NonNullable<NonNullable<NonNullable<AlertInfosQuery['public']>['alert']>['infos']>[number];
+
+type TabKey = string[];
+const keySelector = (info: string) => Number(info);
+
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx]: do we need to convert it to number

File: src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+import { AlertInfosQuery } from '#generated/types/graphql';
+
+import AreaAlertInfo from '../AlertInfo';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type AletInfosType = NonNullable<NonNullable<AlertInfosQuery['public']>['alert']>;
+
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx]: Let's rename it to `AlertDetail`

File: src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+type InfoAlertType = NonNullable<NonNullable<NonNullable<AlertInfosQuery['public']>['alert']>['infos']>[number];
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx]:
```suggestion
type AlertInfo = NonNullable<NonNullable<NonNullable<AlertInfosQuery['public']>['alert']>['infos']>[number];
```

File: src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+    data: AletInfosType;
+}
+
+function AlertDetail(props: AlertProps) {
+    const {
+        data,
+    } = props;
+
+    const isUnknown = data?.admin1s?.map((admin) => admin.isUnknown);
+
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx]:
```suggestion
    const unknownAdmin1Alerts = data?.admin1s?.map((admin) => admin.isUnknown);

```

File: src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+        const newList = listToMap(
+            data?.infos ?? [],
+            (d) => d.id,
+            (d) => d?.language,
+        );
+        setTabKeys(Object.keys(newList));
+        setActiveTab(Object.keys(newList)?.[0]);
+
+        return newList;
+    }, [data?.infos]);
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx]: Let's directly use the `infos` array and generate the tabs

File: src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+            data: value,
+        }),
+        [getTabName],
+    );
+
+    return (
+        <Container
+            className={styles.alertDetails}
+            childrenContainerClassName={styles.content}
+            heading={data?.infos.map((info) => info.event)}
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx]: Let's fetch `info` and use its `headline` field

File: src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+                    </>
+                )}
+                withoutLabelColon
+            />
+            {isUnknown ? (
+                <Button
+                    name={undefined}
+                    variant="primary"
+                />
+            ) : null}
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx]: Let's use `badge` component.

Comment (samshara) [reply src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx]: Let's use && operator

File: src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+            {data?.url ? (
+                <Link
+                    to={data?.url}
+                    className={styles.contactButton}
+                    target="_blank"
+                >
+                    <ShareBoxFillIcon />
+                    {strings.alertOrigin}
+                </Link>
+            ) : null}
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx]: Let's use && operator

File: src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+                    onChange={setActiveTab}
+                    variant="primary"
+                >
+                    <TabList>
+                        {/* TODO: use list for tab */}
+                        {tabKeys?.map((tab, index: number) => (
+                            <Tab key={tab} name={tab}>
+                                {getTabName(index)}
+                            </Tab>
+                        ))}
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/AlertDetail/index.tsx]: Let's use `infos` array to create info tabs

File: src/views/Home/AlertsView/AlertsAside/AlertListItem/index.tsx [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+                    onClick={onCountryClick}
+                    variant="tertiary"
+                >
+                    {data.info?.event}
+                    {' '}
+                    -
+                    {data.info?.category}
+                </Button>
+            )}
+        />
```
Comment (samshara) [inline src/views/Home/AlertsView/AlertsAside/AlertListItem/index.tsx]: Let's show appropriate information

File: src/views/Home/AlertsView/index.tsx [hunk 15]
```diff
@@ -28,12 +41,101 @@ query CountryList {
...
+                  event
+                  category
+                }
+            }
+        limit
+        offset 
+        count
+      }
+    }
+  }
```
Comment (samshara) [inline src/views/Home/AlertsView/index.tsx]: move this to inner component

**Review by samshara — COMMENTED** [review]
Let's refactor the components and logic. Let's work on the AlertList tab first.

**Review by frozenhelium — APPROVED** [review]

**Review by samshara — APPROVED** [review]

**Pull request merged**